araproc/framework/map_utilities.py

```diff
@@ -1,74 +1,296 @@
 import ctypes
+import os
+import numpy as np
 import ROOT
+from functools import partial
+from araproc.framework import constants
 
 def get_corr_map_peak(the_map  = None):
         
-        """
-        Little utility helper function
+    """
+    Little utility helper function
 
-        Parameters
-        ----------
-        the_map : ROOT.TH2D
-            A 2D map you want the peak found
+    Parameters
+    ----------
+    the_map : ROOT.TH2D
+        A 2D map you want the peak found
         
         
-        Returns
-        peak_corr : float
-            The peak in the correlation  map
-        peak_phi : int
-            The theta value of the peak of the correlation map
-        peak_phi : int
-            The phi value of the peak of the correlation map
-        """
+    Returns
+    -------
+    peak_corr : float
+        The peak in the correlation  map
+    peak_phi : int
+        The theta value of the peak of the correlation map
+    peak_phi : int
+        The phi value of the peak of the correlation map
+    """
         
-        if the_map is None:
-            raise Exception("No TH2D map was passed")
+    if the_map is None:
+       raise Exception("No TH2D map was passed")
     
-        _peakZ = ctypes.c_int()
-        _peakTheta = ctypes.c_int()
-        _peakPhi = ctypes.c_int()
-        the_map.GetMaximumBin(_peakPhi, _peakTheta, _peakZ)
+    _peakZ = ctypes.c_int()
+    _peakTheta = ctypes.c_int()
+    _peakPhi = ctypes.c_int()
+    the_map.GetMaximumBin(_peakPhi, _peakTheta, _peakZ)
 
-        peak_corr = the_map.GetMaximum()
-        peak_phi = the_map.GetXaxis().GetBinCenter(_peakPhi.value)
-        peak_theta = the_map.GetYaxis().GetBinCenter(_peakTheta.value)
-        return peak_corr, peak_phi, peak_theta
+    peak_corr = the_map.GetMaximum()
+    peak_phi = the_map.GetXaxis().GetBinCenter(_peakPhi.value)
+    peak_theta = the_map.GetYaxis().GetBinCenter(_peakTheta.value)
+    return peak_corr, peak_phi, peak_theta
 
-def calculate_avg_antenna_xyz(station_id, num_channels):
```

Comment:
I think it's okay to get ride of this if we want, but let's make sure it's not being used anywhere else in the code. In fact, maybe it's best to make this a wrapper function or something to ensure backwards compatibility.

Reply:
Right outside the class. Probably it would just define an instance of the class the only exists inside the function or something